tool/src/demo_gen/terminus.rs

```diff
@@ -0,0 +1,380 @@
+        let out_param = |type_name| {
+            // This only works for enums, since otherwise we break the type into its component parts.
+            let label = attrs
+                .and_then(|attrs| {
+                    let label = attrs
+                        .input_cfg
+                        .get(&param_name)
+                        .map(|cfg| cfg.label.clone())
+                        .unwrap_or_default();
+
+                    if label.is_empty() {
+                        None
+                    } else {
+                        Some(label)
+                    }
+                })
+                .unwrap_or(heck::AsUpperCamelCase(param_name.clone()).to_string());
+
+            let mut param_info = ParamInfo {
+                js: param_name,
+                label,
+                type_name,
+            };
+
+            self.terminus_info.params.push(param_info.clone());
+
+            // Grab arguments without having to name them
+            param_info.js = format!("terminusArgs[{}]", self.terminus_info.params.len() - 1);
+            node.params.push(param_info);
+        };
+
+        // TODO: I think we need to check for struct and opaque types as to whether or not these have attributes that label them as provided as a parameter.
+        match param_type {
+            Type::Primitive(_) | Type::Enum(_) | Type::Slice(_) => {
+                let type_name = match param_type {
+                    Type::Primitive(p) => self.formatter.fmt_primitive_as_ffi(*p).to_string(),
+                    Type::Slice(hir::Slice::Str(..)) => self.formatter.fmt_string().to_string(),
+                    Type::Slice(hir::Slice::Primitive(_, p)) => {
+                        self.formatter.fmt_primitive_list_type(*p).to_string()
+                    }
+                    Type::Slice(hir::Slice::Strs(..)) => "Array<String>".to_string(),
+                    Type::Enum(e) => {
+                        let type_name = self.formatter.fmt_type_name(e.tcx_id.into()).to_string();
+
+                        if e.resolve(self.tcx).attrs.disable {
+                            self.errors
+                                .push_error(format!("Found usage of disabled type {type_name}"))
+                        }
+
+                        type_name
+                    }
+                    _ => unreachable!("Unknown primitive type {:?}", param_type),
+                };
+
+                out_param(type_name);
+            }
+            Type::Opaque(o) => {
+                let op = o.resolve(self.tcx);
+                let type_name = self.formatter.fmt_type_name(o.tcx_id.into());
+
+                let all_attrs = &o.resolve(self.tcx).attrs;
+                if all_attrs.disable {
+                    self.errors
+                        .push_error(format!("Found usage of disabled type {type_name}"))
+                }
+                let attrs = &all_attrs.demo_attrs;
+
+                if attrs.external {
```

Comment:
question: how does this work? I guess I'm not clear on what "is not evaluated but is passed to the rendering system to provide" means for demos.